### PR TITLE
chore: update golangci-lint to v1.61.0

### PR DESCRIPTION
**Pull request by blkperl**
### Motivation

* A newer version of golangci-lint is requried to support golang 1.23

### Modifications

Changed `exportloopref` to `copyloopvar`

Fixed the following lint errors
* `printf: non-constant format string in call to github.com/sirupsen/logrus.Infof`
* `printf: non-constant format string in call to (*testing.common).Fatalf (govet)`
* `printf: non-constant format string in call to fmt.Sprintf (govet)`
* `printf: non-constant format string in call to google.golang.org/grpc/status.Errorf (govet)`
* `suite-broken-parallel: testify v1 does not support suite's parallel tests and subtests`
* `suite-subtest-run: use s.Run to run subtest (testifylint)`
* `The copy of the 'for' variable "limit" can be deleted (Go 1.22+) (copyloopvar)`

### Verification

Lint passes and existing test coverage

### CODE REVIEW

**Comment by isubasinghe** [comment]
I believe this fixes memory usage issues, currently occasionally hit 84gb of memory usage before OOM.

**Comment by Joibel** [comment]
> I believe this fixes memory usage issues, currently occasionally hit 84gb of memory usage before OOM.

Which part fixes that?

**Comment by isubasinghe** [comment]
> Which part fixes that?

The newer version of golangci-lint

**Comment by Joibel** [comment]
> > Which part fixes that?
> 
> The newer version of golangci-lint

Oh, during linting, not at runtime. More coffee needed. Ok - I remember you saying it did mad things for you.

File: test/e2e/cron_test.go [hunk 1]
```diff
@@ -43,7 +43,6 @@ func (s *CronSuite) TearDownSuite() {
 
 func (s *CronSuite) TestBasic() {
 	s.Run("TestBasic", func() {
-		s.T().Parallel()
```
Comment (Joibel) [inline test/e2e/cron_test.go:46]: These don't do anything, and as this was the only attempt in the code to parallelise tests, we could remove the `-parallel` flag on the test suite.

I'd like to find a way to reintroduce parallel testing in the future though, so let's not do that.

**Review by Joibel — APPROVED** [review]

**Pull request merged**